Configuration.props

```diff
@@ -14,10 +14,14 @@
   <PropertyGroup>
     <ProductVersion>8.3.99</ProductVersion>
     <!-- *Latest* API level binding that we support; used when building src/Xamarin.Android.Build.Tasks -->
-    <AndroidLatestApiLevel Condition="'$(AndroidLatestApiLevel)' == ''">27</AndroidLatestApiLevel>
-    <AndroidLatestFrameworkVersion Condition="'$(AndroidLatestFrameworkVersion)' == ''">v8.1</AndroidLatestFrameworkVersion>
+    <AndroidLatestStableApiLevel Condition="'$(AndroidLatestStableApiLevel)' == ''">27</AndroidLatestStableApiLevel>
+    <AndroidLatestStablePlatformId Condition="'$(AndroidLatestStablePlatformId)' == ''">$(AndroidLatestStableApiLevel)</AndroidLatestStablePlatformId>
+    <AndroidLatestApiLevel Condition="'$(AndroidLatestApiLevel)' == ''">28</AndroidLatestApiLevel>
+    <AndroidLatestPlatformId Condition=" '$(AndroidLatestPlatformId)' == '' ">P</AndroidLatestPlatformId>
+    <AndroidLatestFrameworkVersion Condition="'$(AndroidLatestFrameworkVersion)' == ''">v8.1.99</AndroidLatestFrameworkVersion>
     <!-- The API level and TargetFrameworkVersion for the default Mono.Android.dll build -->
-    <AndroidApiLevel Condition=" '$(AndroidApiLevel)' == '' ">$(AndroidLatestApiLevel)</AndroidApiLevel>
+    <AndroidApiLevel Condition=" '$(AndroidApiLevel)' == '' ">$(AndroidLatestStableApiLevel)</AndroidApiLevel>
+    <AndroidPlatformId Condition=" '$(AndroidPlatformId)' == '' ">$(AndroidLatestStablePlatformId)</AndroidPlatformId>
     <AndroidFrameworkVersion Condition=" '$(AndroidFrameworkVersion)' == '' ">$(AndroidLatestFrameworkVersion)</AndroidFrameworkVersion>
```

Comment:
With that background out of the way, I quite like https://github.com/xamarin/xamarin-android/pull/1380/commits/fac54aeb2ef09620104b0555acfa0d946af030b1, separating "stable" from "latest."

There remains one problem, though: [`src/Mono.Android/Mono.Android.csproj` uses `$(AndroidFrameworkVersion)` for it's output path](https://github.com/xamarin/xamarin-android/blob/6436b11fdadf305d6c3b62fd4744dae839929310/src/Mono.Android/Mono.Android.csproj#L37):

```xml
<OutputPath>$(XAInstallPrefix)xbuild-frameworks\MonoAndroid\$(AndroidFrameworkVersion)\</OutputPath>
```

This can't really be changed, in that I'm not sure what you could change it *to*. `Mono.Android.csproj` needs to be separately built for each API Level/Id/Framework version, and `$(AndroidApiLevel)`/`$(AndroidPlatformId)`/`$(AndroidFrameworkVersion)` control those values.

Thus, [the problem](https://github.com/xamarin/xamarin-android/pull/1380/files#diff-45dd9701c8003d3104cd2ffb3d37f3d2R25):

```xml
<AndroidFrameworkVersion Condition=" '$(AndroidFrameworkVersion)' == '' ">$(AndroidLatestFrameworkVersion)</AndroidFrameworkVersion>
```

`$(AndroidFrameworkVersion)` is `$(AndroidLatestFrameworkVersion)`, which means the default `src/Mono.Android.csproj` build will create the `v8.1.99` framework, *not* `v8.1`. Worse, it'll be for the `$(AndroidLatestStableApiLevel)` (27) binding.

Which is [precisely what we see in the build log](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/2660/console):

```
Copying file from '/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/src/Mono.Android/obj/Debug/android-27/Mono.Android.dll' to '/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/bin/Debug/lib/xamarin.android/xbuild-frameworks/MonoAndroid/v8.1.99/Mono.Android.dll'
```

This line should be changed to:

```xml
<AndroidFrameworkVersion Condition=" '$(AndroidFrameworkVersion)' == '' ">$(AndroidLatestStableFrameworkVersion)</AndroidFrameworkVersion>
```

This in turn means we need to add a `$(AndroidLatestStableFrameworkVersion)` value.

This is also more-or-less why the [**macOS+xbuild PR Build** is failing](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/2660/console): `$(AndroidApiLevel)` is 27, while `$(AndroidFrameworkVersion)` is v8.1.99. They're inconsistent. The result is in the `System.EnterpriseServices` build:

```
Task "ResolveSdks"
        Using task ResolveSdks from Xamarin.Android.Tasks.ResolveSdks, Xamarin.Android.Build.Tasks, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
        ResolveSdksTask:
          AndroidApiLevel: 27
          AndroidSdkBuildToolsVersion: 27.0.3
          AndroidSdkPath: /Users/builder/android-toolchain/sdk
          AndroidNdkPath: /Users/builder/android-toolchain/ndk
          JavaSdkPath: 
          ReferenceAssemblyPaths: 
            /Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/bin/Debug/lib/xamarin.android/xbuild-frameworks/MonoAndroid/v1.0/
          TargetFrameworkVersion: v8.1.99
...
System.EnterpriseServices/System.EnterpriseServices.csproj: error XA0001: Unsupported or invalid $(TargetFrameworkVersion) value of 'v8.1'. Please update your Project Options.
```

`$(AndroidApiLevel)` is preferred over `$(TargetFrameworkVersion)`, so by setting it to 27 the `<ResolveSdks/>` task would (normally) attempt to find the v8.1 framework. However, because of `$(AndroidFrameworkVersion)`, there is no v8.1 framework, so the lookup fails, and everything breaks.

Reply:
Thanks for the investigation on that error. I have added that new framework variable for stable API. Let's see if it builds...